clang/compiler-course/Kholin_K_prefixes/Kholin_K_prefixes_plugin.cpp

```diff
@@ -0,0 +1,124 @@
+#include "clang/AST/ASTConsumer.h"
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Frontend/FrontendPluginRegistry.h"
+#include "clang/Rewrite/Core/Rewriter.h"
+#include "llvm/Support/raw_ostream.h"
+#include <fstream>
+#include <optional>
+
+using namespace clang;
+
+class FindNamedClassVisitor
+    : public RecursiveASTVisitor<FindNamedClassVisitor> {
+public:
+  explicit FindNamedClassVisitor(ASTContext *Context, Rewriter &R)
+      : Context(Context), Rewrite(R) {}
+
+  std::optional<std::string> getVarPrefix(const VarDecl *Decl) {
+    if (Decl->isFileVarDecl() && !Decl->isStaticLocal()) {
+      return "global_";
+    } else if (Decl->isStaticLocal()) {
+      return "static_";
+    } else if (Decl->isLocalVarDecl() && !isa<ParmVarDecl>(Decl)) {
+      return "local_";
+    } else if (isa<ParmVarDecl>(Decl)) {
+      return "param_";
+    }
```

Comment:
Just a nit: checks may be a bit simplified if swapped:
```suggestion
    if (Decl->isStaticLocal()) {
      return "static_";
    } else if (Decl->isFileVarDecl()) {
      return "global_";
    } else if (isa<ParmVarDecl>(Decl)) {
      return "param_";
    } else if (Decl->isLocalVarDecl()) {
      return "local_";
    }
```

Reply:
Fixed! Yeah, it makes more sense that way.

Reply:
Sorry, I don't see this fix in source code.

Reply:
I forgot about it when I fixed another bug. One moment...

Reply:
Fixed!